### PR TITLE
QOL for Development with Citation Manager

### PR DESCRIPTION
**Pull request by le-sun**
Calling the pdf callback endpoint manually after a pdf is uploaded with the presigned url
 - Only happens in development

### CODE REVIEW

File: components/ReferenceManager/references/ReferencesContainer.tsx [hunk 1]
```diff
@@ -154,10 +158,31 @@ function ReferencesContainer({
...
       });
-      const result = fetch(preSignedUrl, {
-        method: "PUT",
-        body: fileBlob,
-      });
+
+      if (isDevelopment) {
+        const preSignedUrlParts = preSignedUrl.split("?AWS");
+        const path = preSignedUrlParts[0].split(".com/")[1];
+        const result = await fetch(preSignedUrl, {
```
[the remark this answers is not in the document]
Comment (le-sun) [reply components/ReferenceManager/references/ReferencesContainer.tsx]: @lightninglu10 Updated

File: config/api.js [hunk 2]
```diff
@@ -1301,6 +1301,9 @@ const routes = (BASE_URL) => {
         );
       }
     },
+    CITATION_UPLOAD_CALLBACK: () => {
+      return BASE_URL + `citation_entry/upload_pdfs_callback/`;
```
[the remark this answers is not in the document]
Comment (le-sun) [reply config/api.js]: Updated

**Review by calvinhlee23 — APPROVED** [review]
lgtm

File: components/ReferenceManager/references/ReferencesContainer.tsx [hunk 3]
```diff
@@ -154,10 +158,27 @@ function ReferencesContainer({
...
+
       const result = fetch(preSignedUrl, {
         method: "PUT",
         body: fileBlob,
       });
+
+      if (isDevelopment) {
+        const preSignedUrlParts = preSignedUrl.split("?AWS");
+        const path = preSignedUrlParts[0].split(".com/")[1];
+        await result;
```
[the remark this answers is not in the document]
Comment (le-sun) [reply components/ReferenceManager/references/ReferencesContainer.tsx:163]: Yeah, the pre-signed url seems to always include certain elements that we can just cut off to get the path of the file.

**Pull request merged**